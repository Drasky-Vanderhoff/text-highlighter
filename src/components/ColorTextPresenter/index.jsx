import React from 'react'
import PropTypes from 'prop-types'
import './index.scss'

const ColorTextPresenter = ({
  colorTextItemsLists
}) => (
  <blockquote className="color-text-presenter">
    {colorTextItemsLists.map(
        ({color, subtext}, idx) => <span key={idx}>
          <mark style={{backgroundColor: color}}>
            {subtext}
          </mark>
          <br/>
        </span>
    )}
  </blockquote>
)

export const ColorTextPresenterPropTypes = {
  colorTextItemsLists: PropTypes.arrayOf(
    PropTypes.shape({
      color: PropTypes.string,
      subtext: PropTypes.string
    })
  )
}

ColorTextPresenter.propTypes = ColorTextPresenterPropTypes

ColorTextPresenter.defaultProps = {
  colorTextItemsLists: []
}

export default ColorTextPresenter