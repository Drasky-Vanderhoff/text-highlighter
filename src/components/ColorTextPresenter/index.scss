.color-text-presenter {
  left: 10%;
  width: 80%;
  height: 250px;
  border: 2px solid black;
  text-align: left;
}