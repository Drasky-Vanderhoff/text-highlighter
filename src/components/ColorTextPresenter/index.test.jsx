import React from 'react'
import ColorTextPresenter from './'
import renderer from 'react-test-renderer'

describe('ColorTextPresenter Component', () => {
  test('should render with default values', () => {
    expect(
      renderer.create(<ColorTextPresenter />).toJSON()
    ).toMatchSnapshot()
  })

  test('should render the color text items in the list', () => {
    const props = {
      colorTextItemsLists: [
        {
          color: "red",
          subtext: "lorem"
        },
        {
          color: "blue",
          subtext: "ip"
        },
        {
          color: "blue",
          subtext: "sum"
        },
        {
          color: "yellow",
          subtext: "more text just because"
        }
      ]
    }
    expect(
      renderer.create(<ColorTextPresenter {...props}/>).toJSON()
    ).toMatchSnapshot()
  })
})
