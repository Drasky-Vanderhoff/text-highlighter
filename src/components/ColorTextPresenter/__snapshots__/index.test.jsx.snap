// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ColorTextPresenter Component should render the color text items in the list 1`] = `
<blockquote
  className="color-text-presenter"
>
  <span>
    <mark
      style={
        Object {
          "backgroundColor": "red",
        }
      }
    >
      lorem
    </mark>
    <br />
  </span>
  <span>
    <mark
      style={
        Object {
          "backgroundColor": "blue",
        }
      }
    >
      ip
    </mark>
    <br />
  </span>
  <span>
    <mark
      style={
        Object {
          "backgroundColor": "blue",
        }
      }
    >
      sum
    </mark>
    <br />
  </span>
  <span>
    <mark
      style={
        Object {
          "backgroundColor": "yellow",
        }
      }
    >
      more text just because
    </mark>
    <br />
  </span>
</blockquote>
`;

exports[`ColorTextPresenter Component should render with default values 1`] = `
<blockquote
  className="color-text-presenter"
/>
`;
